lib/nokogiri/xml/node.rb

```diff
@@ -192,16 +198,47 @@ def decorate!
 
       # :section: Manipulating Document Structure
 
-      ###
-      # Add +node_or_tags+ as a child of this Node.
-      #
-      # +node_or_tags+ can be a Nokogiri::XML::Node, a ::DocumentFragment, a ::NodeSet, or a String
-      # containing markup.
-      #
-      # Returns the reparented node (if +node_or_tags+ is a Node), or NodeSet (if +node_or_tags+ is
-      # a DocumentFragment, NodeSet, or String).
-      #
-      # Also see related method +<<+.
+      # :call-seq:
+      #   add_child(object) -> object
```

Comment:
I've been gradually introducing a documentation convention to give a brief overview of arguments and argument types, which I'd like to ask you to use (or improve upon).

Good examples of this convention are in this file, and applying it here might look like:

```
[Parameters]
- +object+ (String, Node, NodeSet) The Node or Nodes to be added as children of this node. A String will be parsed as a fragment and the resulting NodeSet will be added as children.
```

followed by the prose explanation and examples below. WDYT?

Reply:
What you're suggesting fits well with my own personal philosophy.  (That's how I wrote in various DEC programmer's reference manuals back at the Dawn of Time.)

However, I think you should consider the Ruby[ Documentation Guide](https://docs.ruby-lang.org/en/master/contributing/documentation_guide_md.html), first drafted by @jeremyevans and often discussed and augmented since.  It represents consensus in the Ruby community, which I think has value in itself.  But it also gives a consistent "Feel" to the Ruby documentation that readers might appreciate also finding in the gems' documentation.

But of course, "your call."

Reply:
Do you mean [this section in the Documentation Guide](https://docs.ruby-lang.org/en/master/contributing/documentation_guide_md.html#label-Argument+Description+-28if+necessary-29)? I don't find it specifies much of anything beyond "use a labeled list", but to be clear I'm fine with that if that's what you want to go with.

The general principal is: give a brief summary of arguments and their expected types. I'm not finicky about the particulars of the style we use to present it.

Reply:
Also, the "if necessary" bit in that section makes me groan a bit. I get it, we don't need to over-specify things for simple methods with simple arguments.

But all of these Node methods in particular take complex arguments (could be a Node, could be a NodeSet, could be a String that is cast as a NodeSet via fragment parsing) and so, yes, I think for those kinds of methods, documenting the arguments is necessary. WDYT?